demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_payload_operations.c

```diff
@@ -0,0 +1,372 @@
+    {
+        /* Parameters in this example is length 1. */
+        cborRet = cbor_encoder_create_map( &mapEncoder, &parametersEncoder, 1 );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encode_text_stringz( &parametersEncoder, "SerialNumber" );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encode_text_string( &parametersEncoder, pSerial, serialLength );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encoder_close_container( &mapEncoder, &parametersEncoder );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encoder_close_container( &encoder, &mapEncoder );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        status = true;
+        *pOutLengthWritten = cbor_encoder_get_buffer_size( &encoder, ( uint8_t * ) pBuffer );
+    }
+    else
+    {
+        LogError( ( "Error during CBOR encoding: %s", cbor_error_string( cborRet ) ) );
+
+        if( ( cborRet & CborErrorOutOfMemory ) != 0 )
+        {
+            LogError( ( "Cannot fit RegisterThing request payload into buffer." ) );
+        }
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+bool parseCsrResponse( const char * pResponse,
+                       size_t length,
+                       char * pCertificate,
+                       size_t * pCertificateLength,
+                       char * pCertificateId,
+                       size_t * pCertificateIdLength,
+                       char * pOwnershipToken,
+                       size_t * pOwnershipTokenLength )
+{
+    bool status = false;
+    CborError cborRet;
+    CborParser parser;
+    CborValue map;
+    CborValue value;
+
+    /* For details on the CreateCertificatefromCsr response payload format, see:
+     * https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html#register-thing-response-payload
+     */
+    cborRet = cbor_parser_init( ( const uint8_t * ) pResponse, length, 0, &parser, &map );
+
+    if( cborRet != CborNoError )
+    {
+        LogError( ( "Error initializing parser for CreateCertificateFromCsr response: %s.", cbor_error_string( cborRet ) ) );
+    }
+    else if( !cbor_value_is_map( &map ) )
+    {
+        LogError( ( "CreateCertificateFromCsr response not a map type." ) );
+    }
+    else
+    {
+        cborRet = cbor_value_map_find_value( &map, "certificatePem", &value );
+
+        if( cborRet != CborNoError )
+        {
+            LogError( ( "Error searching CreateCertificateFromCsr response: %s.", cbor_error_string( cborRet ) ) );
+        }
+        else if( value.type == CborInvalidType )
+        {
+            LogError( ( "\"certificatePem\" not found in CreateCertificateFromCsr response." ) );
+        }
+        else if( value.type != CborTextStringType )
+        {
+            LogError( ( "\"certificatePem\" is an unexpected type in CreateCertificateFromCsr response." ) );
```

Comment:
```suggestion
            LogError( ( "Value for \"certificatePem\" entry is not of the expected text string type in CreateCertificateFromCsr response payload." ) );
```

Reply:
Same suggestion for other logs in the function to mention about the expected object type for the field's value